### PR TITLE
Jigsaw UX Changes

### PR DESCRIPTION
**Pull request by ryansloan**
Responding to frequent user feedback and classroom observations, making a couple small tweaks to Jigsaw puzzles:

- **removing the pink "loop-like" block from the levels** - students don't learn loops until later, and we never actually explain how to use this block in Jigsaw.

- **Numbering the color-matching blocks** These are hard to match to the partially transparent preview on bad screens, and could also present problems for students with color acuity issues or colorblindness

Numbered Jigsaw blocks

### CODE REVIEW

File: apps/src/jigsaw/blocks.js [hunk 1]
```diff
@@ -172,13 +171,16 @@ exports.install = function (blockly, blockInstallOptions) {
...
 
-function generateBlankBlock(blockly, skin, name, hsv, width, hasAppend) {
+function generateBlankBlock(blockly, skin, name, hsv, width, hasAppend, label) {
   blockly.Blocks[name] = {
     helpUrl: '',
     init: function () {
       this.setHSV.apply(this, hsv);
       this.appendDummyInput()
-        .appendTitle(new blockly.FieldImage(skin.blank, width, 54));
+        .appendTitle(new blockly.FieldImage(skin.blank, (width-8)/2, 54))
```
Comment (joshlory) [inline apps/src/jigsaw/blocks.js]: Why the change of `width` -> `(width-8)/2`?

Tiny nit: calculate this once above, and store it in a descriptive variable.

File: apps/src/jigsaw/blocks.js [hunk 2]
```diff
@@ -172,13 +171,16 @@ exports.install = function (blockly, blockInstallOptions) {
...
+        .appendTitle(new blockly.FieldLabel(label,
+            {fixedSize: {width: (width-8)/2, height: 64}, fontSize:32}))
```
Comment (joshlory) [inline apps/src/jigsaw/blocks.js]: Tiny nit: `fontSize: 32` (space after the colon).

File: apps/src/jigsaw/blocks.js (hunk 1 above)
Comment (ryansloan) [reply apps/src/jigsaw/blocks.js]: Previously, `width` was used to supply the total width of the block, but this is only accounting half of it, but this function is only used here, so I can switch to just passing in a smaller value for width if that would be clearer

Comment (joshlory) [reply apps/src/jigsaw/blocks.js]: Yeah, I think it's a little cleaner to update the places where we call `generateBlankBlock` above with the new value.

Comment (joshlory) [reply apps/src/jigsaw/blocks.js]: Or just inline it as always 36 and remove the parameter, now that it's always the same value!

File: apps/src/jigsaw/blocks.js [hunk 3]
```diff
@@ -172,13 +171,16 @@ exports.install = function (blockly, blockInstallOptions) {
...
+        .setAlign(Blockly.ALIGN_CENTRE);
       this.setPreviousStatement(true);
       if (hasAppend) {
```
Comment (joshlory) [inline apps/src/jigsaw/blocks.js]: Looks like this can go away now 🎉

**Comment by jeremydstone** [comment]
Congrats on your first PR @ryansloan!

**Comment by breville** [comment]
Great work, mysterious contributor @ryansloan.
